### PR TITLE
Add colors to error message hints 

### PR DESCRIPTION
**Pull request by ofonime41**
1. Created a `hint` style tag and define the styles for the `Hint: `prefix.
2. Added the` hint` tag to all the Hint: prefix.
3. Fixed a typographical error.

### CODE REVIEW

**Comment by ofonime41** [comment]
This is a reopen of pull request [11672](https://github.com/ocaml/ocaml/pull/11672).
Couldn't reopen it because the branch was recreated. 

Kindly review @Octachron. Thank you.

File: tools/ocamltex.ml [hunk 1]
```diff
@@ -765,7 +765,7 @@ let process_file file =
          spanned the interval %d-%d,@ \
          intersecting with another \"%a\" transform @ \
          on the %d-%d interval.@ \
-         Hind: did you try to elide a code fragment which raised a warning?"
+         @{<hint>Hint@}: did you try to elide a code fragment which raised a warning?"
```
Comment (cfcs) [inline tools/ocamltex.ml]: Well done for spotting the spelling mistake too!

File: utils/misc.mli [hunk 2]
```diff
@@ -374,7 +374,6 @@ module Color : sig
     | BG of color (* background *)
     | Bold
     | Reset
-
```
Comment (cfcs) [inline utils/misc.mli:377]: The blank line deleted here seems like an accident. It is not hugely important, but it makes it harder to track the provenance of code using `git blame`, so if you can reinstate the blank line that would be great. :-)

**Comment by cfcs** [comment]
It looks like you missed a few in `utils/warnings.ml`?
You can use `git grep SOMETHING`  to look for `SOMETHING`:
```shell
$ git grep Hint:
utils/warnings.ml:            "@ @[Hint: Did you make a spelling mistake \
utils/warnings.ml:           @[Hint: Enabling or disabling a warning by its mnemonic name \
utils/warnings.ml:       Hint: Did you forget spaces when writing the infix operator `( * )'?"
utils/warnings.ml:         Hint: Did you mean 'type %s = unit'?" name
```
(press `q` to get out of `less`, the "pager", if you get stuck in the terminal showing the matching text).

**Comment by Octachron** [comment]
The warning printer is currently printing string messages directly without taking in account any formatter settings. It is best to leave those hints for a second PR.

File: tools/ocamltex.ml (hunk 1 above)
Comment (ofonime41) [reply tools/ocamltex.ml]: Thank you.

File: typing/typecore.ml [hunk 3]
```diff
@@ -5716,7 +5717,7 @@ let report_error ~loc env = function
         if b then
           fprintf ppf ".@.@[<hov>%s@ %s@ %s@]"
             "This simple coercion was not fully general."
-            "Hint: Consider using a fully explicit coercion"
+            "@{<hint>Hint@}: Consider using a fully explicit coercion"
```
Comment (Octachron) [inline typing/typecore.ml]: This hint needs to be inlined inside the format string, otherwise it will not be formatted.

File: Changes [hunk 4]
```diff
@@ -134,9 +134,14 @@ Working version
...
   (Pierre Chambart and Vincent Laviron, review by Mark Shinwell)
 
+
+- #11646: Add colors to error message hints.
+  (Christiana Anthony, review by Florian Angeletti)
+
 - #11653: Add the -no-absname option to ocamlc, ocamlopt and ocamldep.
   (Abiola Abdulsalam, review by Sébastien Hinderer and Florian Angeletti)
 
+
```
Comment (Octachron) [inline Changes]: Another line that can be removed.

File: typing/printtyp.ml [hunk 5]
```diff
@@ -172,15 +172,15 @@ module Conflicts = struct
       | [namespace, a] ->
           Format.fprintf ppf
         "@ \
-         @[<2>Hint: The %a %s has been defined multiple times@ \
+         @[<2>@{<hint>Hint@}: The %a %s has been defined multiple times@ \
```
Comment (Octachron) [inline typing/printtyp.ml:175]: Self-note: this formatting does not work in the ambiguous constructor disambiguation due to the
use of `asprintf`. This should be fixed later.

**Comment by Octachron** [comment]
Outside of the remark on the full coercion hint, the PR looks good to me.

There are few non-tagged hints remaining in `utils/warning.ml` which requires more expertise and could be converted at a later date.

The color blue seems neutral enough for a hint and fit within the
- error : red
-  warning : magenta
-  hint : blue

hierarchy. Moreover, we could change it to a bold face with a single line change.

File: typing/printtyp.ml (hunk 5 above)
Comment (ofonime41) [reply typing/printtyp.ml:175]: Thank you for the reviews.
I'll make the necessary changes needed for this PR right away

**Comment by ofonime41** [comment]
> 
Screenshot_20221103_230325

The hint style is colored blue and bold face. Are you suggesting I remove the color and make it just bold?

**Review by Octachron — APPROVED** [review]
The PR is ready to merge for me.

**Pull request merged**